src/state.rs

```diff
@@ -54,13 +54,14 @@ pub enum Trans<T, E> {
     /// if there are none.
     Pop,
     /// Pause the active state and push a new state onto the stack.
-    Push(Box<State<T, E>>),
+    Push(Box<State<T, E> + Send + Sync>),
```

Comment:
That would disallow dispatchers in states.

Reply:
Indeed.